src/common_labsoft/components/http_client.clj

```diff
@@ -0,0 +1,83 @@
+(ns common-labsoft.components.http-client
+  (:require [common-labsoft.protocols.config :as protocols.config]
+            [common-labsoft.protocols.http-client :as protocols.http-client]
+            [common-labsoft.protocols.token :as protocols.token]
+            [clj-http.client :as client]
+            [cheshire.core :as cheshire]))
+
+(defn- resolve-replace-map
+  [template-str replace-map]
+  (reduce-kv (fn [s k v] (clojure.string/replace s (str k) (str v))) template-str replace-map))
+
+(defn render-route
+  "Resolves to a full url"
+  ([hosts service-host endpoint replace-map]
+  (let [service (get hosts service-host)
+        endpoints (get service :endpoints)]
+    (resolve-replace-map (str (get service :host) (get endpoints endpoint)) replace-map)))
+  ([hosts service-host endpoint]
+    (render-route hosts service-host endpoint {})))
+
+(defn render-keyworded-url
+  "Resolves :customers/one-customer to its url"
+  [hosts key replace-map]
+  (let [service-host (keyword (namespace key))
+        endpoint (keyword (name key))]
+    (render-route hosts service-host endpoint replace-map)))
+
+(defn resolve-token
+  "Receives a token. If its valid, returns it. If its not, generate a valid token and return it."
+  [token-component token service-name service-password]
+  (let [auth-payload {:auth/service-name service-name :auth/auth-password service-password}
```

Comment:
Let's use `:auth/service` and `:auth/password`. The namespace already contais info sufficient abou what this does